os/board/common/partitions.c

```diff
@@ -34,6 +34,15 @@
 #include <tinyara/fs/mksmartfs.h>
 #endif
 
+#if defined(CONFIG_LIB_BOARDCTL) && defined(CONFIG_BOARDCTL_RESET)
+#include <sys/boardctl.h>
+#endif
+
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
```

Comment:
How about moving these lines into the line from 37 to 39?
Because, reboot reason is used only when we need to reset the board.

Reply:
ok good opinion.